client/src/main/java/org/spine3/base/ListStringifier.java

```diff
@@ -0,0 +1,201 @@
+ *    final List<Integer> convertedList = listStringifier.fromString(stringToConvert);
+ * }
+ *
+ * @param <T> the type of the elements in the list.
+ */
+class ListStringifier<T> extends Stringifier<List<T>> {
+
+    private static final char DEFAULT_ELEMENT_DELIMITER = ',';
+
+    /**
+     * The delimiter for the passed elements in the {@code String} representation,
+     * {@code DEFAULT_ELEMENT_DELIMITER} by default.
+     */
+    private final char delimiter;
+    private final Class<T> listGenericClass;
+    private final String elementDelimiterPattern;
+    private final Escaper escaper;
+
+    /**
+     * Creates a {@code ListStringifier}.
+     *
+     * <p>The {@code DEFAULT_ELEMENT_DELIMITER} is used for element
+     * separation in {@code String} representation of the {@code List}.
+     *
+     * @param listGenericClass the class of the list elements
+     */
+    ListStringifier(Class<T> listGenericClass) {
+        super();
+        this.listGenericClass = listGenericClass;
+        this.delimiter = DEFAULT_ELEMENT_DELIMITER;
+        escaper = Stringifiers.QuotedItem.createEscaper(delimiter);
+        elementDelimiterPattern = createElementDelimiterPattern(delimiter);
+    }
+
+    /**
+     * Creates a {@code ListStringifier}.
+     *
+     * <p>The specified delimiter is used for element separation
+     * in {@code String} representation of the {@code List}.
+     *
+     * @param listGenericClass the class of the list elements
+     * @param delimiter        the delimiter for the passed elements via string
+     */
+    ListStringifier(Class<T> listGenericClass, char delimiter) {
+        super();
+        this.listGenericClass = listGenericClass;
+        this.delimiter = delimiter;
+        escaper = Stringifiers.QuotedItem.createEscaper(delimiter);
+        elementDelimiterPattern = createElementDelimiterPattern(delimiter);
+    }
+
+    private static String createElementDelimiterPattern(char delimiter) {
+        return Pattern.compile("(?<!\\\\)\\\\\\" + delimiter)
+                      .pattern();
+    }
+
+    @Override
+    protected String toString(List<T> list) {
+        final Function<T, QuotedListItem<T>> function = new Function<T, QuotedListItem<T>>() {
+            @Nullable
+            @Override
+            public QuotedListItem<T> apply(@Nullable T input) {
+                if(input == null){
+                    throw newIllegalArgumentException("The list element cannot be null.");
+                }
+                return of(input);
+            }
+        };
+        final List<QuotedListItem<T>> quotedItems = Lists.transform(list, function);
+        final String result = Joiner.on(delimiter)
+                                    .join(quotedItems);
+        return result;
+    }
+
+    @Override
+    protected List<T> fromString(String s) {
+        final String escapedString = escaper.escape(s);
+        final Splitter splitter = Splitter.onPattern(elementDelimiterPattern);
```

Comment:
Looks like this instance of `splitter` can also be created once and forever for this class — as long as `elementDelimiterPattern` is a `final` field.